client/my-sites/earn/components/add-edit-coupon-modal/test/index.jsx

```diff
@@ -0,0 +1,353 @@
+import Modal from 'react-modal';
+import { Provider } from 'react-redux';
+import configureStore from 'redux-mock-store';
+import {
+	getconnectedAccountDefaultCurrencyForSiteId,
+	getconnectedAccountMinimumCurrencyForSiteId,
+} from '../../../../../state/memberships/settings/selectors';
+import {
+	COUPON_DISCOUNT_TYPE_AMOUNT,
+	COUPON_DISCOUNT_TYPE_PERCENTAGE,
+	COUPON_DURATION_3_MONTHS,
+	COUPON_DURATION_FOREVER,
+} from '../../../memberships/constants';
+import RecurringPaymentsCouponAddEditModal from '../index';
+
+const testCoupon1 = {
+	ID: 1,
+	coupon_code: 'COUPON1',
+	cannot_be_combined: false,
+	can_be_combined: true, // TODO: remove after backend migration to 'cannot_be_combined';
+	first_time_purchase_only: false,
+	start_date: '2023-12-25',
+	end_date: '2024-10-23',
+	plan_ids_allow_list: [],
+	limit_per_user: 0,
+	use_duration: false,
+	use_email_allow_list: true,
+	duration: COUPON_DURATION_FOREVER,
+	email_allow_list: [ '*@*.edu', '*@*.org' ],
+	discount_type: COUPON_DISCOUNT_TYPE_PERCENTAGE,
+	discount_value: 0,
+	discount_percentage: 90,
+	discount_currency: 'USD',
+};
+const testCoupon2 = {
+	ID: 2,
+	coupon_code: 'COUPON2',
+	cannot_be_combined: false,
+	can_be_combined: true, // TODO: remove after backend migration to 'cannot_be_combined';
+	first_time_purchase_only: false,
+	start_date: '2023-12-25',
+	end_date: '2024-10-23',
+	plan_ids_allow_list: [],
+	limit_per_user: 0,
+	use_duration: false,
+	use_email_allow_list: true,
+	duration: COUPON_DURATION_FOREVER,
+	email_allow_list: [ '*@*.edu', '*@*.org' ],
+	discount_type: COUPON_DISCOUNT_TYPE_PERCENTAGE,
+	discount_value: 0,
+	discount_percentage: 90,
+	discount_currency: 'USD',
+};
+const testCoupon3 = {
+	ID: 3,
+	coupon_code: 'COUPON3',
+	cannot_be_combined: true,
+	can_be_combined: false, // TODO: remove after backend migration to 'cannot_be_combined';
+	first_time_purchase_only: true,
+	start_date: '2023-12-25',
+	end_date: '2024-10-23',
+	plan_ids_allow_list: [],
+	limit_per_user: 5,
+	use_duration: true,
+	use_email_allow_list: true,
+	duration: COUPON_DURATION_3_MONTHS,
+	email_allow_list: [],
+	discount_type: COUPON_DISCOUNT_TYPE_AMOUNT,
+	discount_value: 10,
+	discount_percentage: 0,
+	discount_currency: 'USD',
+};
+
+const initialState = {
+	sites: { items: {} },
+	siteSettings: { items: {} },
+	ui: { selectedSiteId: 1 },
+	memberships: {
+		couponList: {
+			items: {
+				1: [ testCoupon1, testCoupon2, testCoupon3 ],
+			},
+		},
+		productsList: {
+			items: {},
+		},
+	},
+};
+const closeDialog = () => {};
+
+function WrappedRecurringPaymentsCouponAddEditModal( props ) {
+	const mockStore = configureStore();
+	const store = mockStore( initialState );
+
+	return (
+		<Provider store={ store }>
+			<RecurringPaymentsCouponAddEditModal { ...props } />
+		</Provider>
+	);
+}
+
+describe( 'RecurringPaymentsCouponAddEditModal', () => {
+	let modalRoot;
+
+	const getRandomButton = () => screen.getByText( 'Random' );
```

Comment:
Similarly to other comments, we should be able to find the button via `getByRole('button', {name: 'Random'})` to make sure that the correct semantics are implemented

Reply:
Thank you! Very helpful to know the priority as well.